### PR TITLE
Make compatible with Elbrus (e2k)

### PR DESCRIPTION
**Pull request by mouseratti**
Trying to build under e2k on Alt Linux. These changes allowed to build `vitastor-osd` target successfully. 
Build for x86_64 still works.

### CODE REVIEW

File: src/cluster_client.cpp [hunk 1]
```diff
@@ -929,6 +929,8 @@ bool cluster_client_t::try_send(cluster_op_t *op, int i)
                     meta_rev = ino_it->second.mod_revision;
             }
             part->op = (osd_op_t){
+                .tv_begin = 0,
+                .tv_end = 0,
```
Comment (mouseratti) [inline src/cluster_client.cpp:933]:
```suggestion
```

File: src/osd_cluster.cpp [hunk 2]
```diff
@@ -41,6 +41,14 @@ void osd_t::init_cluster()
...
                 .pool_id = 1,
                 .pg_num = 1,
+                .clean_count = 0, 
+                .total_count = 0,
+                .epoch = 0, 
+                .reported_epoch = 0,
+                .target_history = std::vector<std::vector<osd_num_t>>{},
+                .all_peers = std::vector<osd_num_t>{},
+                .history_changed = false,
+                .cur_peers = std::vector<osd_num_t>{},
```
Comment (mouseratti) [inline src/osd_cluster.cpp:51]:
```suggestion
```

File: src/osd_cluster.cpp [hunk 3]
```diff
@@ -51,6 +59,7 @@ void osd_t::init_cluster()
                 .scheme = POOL_SCHEME_XOR,
                 .pg_size = 3,
                 .pg_minsize = 2,
+                .parity_chunks = 0,
```
Comment (mouseratti) [inline src/osd_cluster.cpp:62]:
```suggestion
```

File: src/osd_cluster.cpp [hunk 4]
```diff
@@ -668,9 +677,14 @@ void osd_t::apply_pg_config()
                          ? 1 : pool_item.second.pg_size - pool_item.second.parity_chunks,
                     .pool_id = pool_id,
                     .pg_num = pg_num,
+                    .clean_count = 0, 
+                    .total_count = 0,
+                    .epoch = 0,
```
Comment (mouseratti) [inline src/osd_cluster.cpp:682]:
```suggestion
```

File: src/osd_cluster.cpp [hunk 5]
```diff
@@ -668,9 +677,14 @@ void osd_t::apply_pg_config()
...
                     .reported_epoch = pg_cfg.epoch,
                     .target_history = pg_cfg.target_history,
                     .all_peers = std::vector<osd_num_t>(all_peers.begin(), all_peers.end()),
+                    .history_changed = false,
+                    .cur_peers = std::vector<osd_num_t>{},
```
Comment (mouseratti) [inline src/osd_cluster.cpp:687]:
```suggestion
```

File: src/osd_flush.cpp [hunk 6]
```diff
@@ -180,6 +180,9 @@ void osd_t::submit_flush_op(pool_id_t pool_id, pg_num_t pg_num, pg_flush_batch_t
                 op->bs_op = NULL;
                 delete op;
             },
+            .oid = {},
+            .version = 0,
+            .offset=0,
```
Comment (mouseratti) [inline src/osd_flush.cpp:185]:
```suggestion
```

File: src/osd_primary_subops.cpp [hunk 7]
```diff
@@ -559,6 +559,9 @@ void osd_t::submit_primary_stab_subops(osd_op_t *cur_op)
                 {
                     handle_primary_bs_subop(subop);
                 },
+                .oid = {},
+                .version = 0,
+                .offset = 0,
```
Comment (mouseratti) [inline src/osd_primary_subops.cpp:564]:
```suggestion
```

File: src/CMakeLists.txt [hunk 8]
```diff
@@ -41,8 +41,9 @@ macro(install_symlink filepath sympath)
 	install(CODE "message(\"-- Created symlink: ${sympath} -> ${filepath}\")")
 endmacro(install_symlink)
 
+find_library(LIBURING_LIBRARIES NAMES liburing uring REQUIRED)
+
```
Comment (mouseratti) [inline src/CMakeLists.txt:45]:
```suggestion
```

File: src/CMakeLists.txt [hunk 9]
```diff
@@ -41,8 +41,9 @@ macro(install_symlink filepath sympath)
...
 find_package(PkgConfig)
```
Comment (mouseratti) [inline src/CMakeLists.txt:46]:
```suggestion
find_package(PkgConfig)
pkg_check_modules(LIBURING REQUIRED liburing)
```

**Comment by vitalif** [comment]
There were 3 main issues (just for history):
1) ALT linux was lacking `liburing.pc` i.e. pkg-config for liburing - suggested to be solved in the distro or separately during build process
2) Elbrus compiler (lcc) was giving the `#1563: class type not suitable for use with designators` error for some structs initialized with designators but without all initialized fields - can be solved by passing `-std=c++2a` during build process. However, `c++2a` is "C++20 Alpha" so it's recommended to use `c++20` instead of it, but lcc doesn't support `c++20`. So this problem is also suggested to be fixed by passing this option during build process
3) Another common error was `expression must be a pointer to a complete object type для void*` for void* pointer arithmetic - I'll merge fixes of this one because it's UB in C++ and it also doesn't compile in clang even on x86

**Comment by vitalif** [comment]
Ok, I fixed (3) so it now builds with clang and as I understand lcc too. Thanks for the changes, some of them got integrated in a bigger commit :-)

**Pull request closed without merging**